From: "Gomonovych, Vasyl" <gomonovych@gmail.com>
Date: Wed, 29 Nov 2017 00:04:06 +0100
Subject: clk: mvebu: armada-37xx-periph: Use PTR_ERR_OR_ZERO()

Git-commit: 9713588f8cd96069716bd8daf720dc7ff921af89
Patch-mainline: v4.16-rc1
References: fate#326470

Fix ptr_ret.cocci warnings:
drivers/clk/mvebu/armada-37xx-periph.c:362:1-3: WARNING: PTR_ERR_OR_ZERO can be used

Use PTR_ERR_OR_ZERO rather than if(IS_ERR(...)) + PTR_ERR

Generated by: scripts/coccinelle/api/ptr_ret.cocci

Signed-off-by: Vasyl Gomonovych <gomonovych@gmail.com>
Signed-off-by: Stephen Boyd <sboyd@codeaurora.org>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/clk/mvebu/armada-37xx-periph.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/drivers/clk/mvebu/armada-37xx-periph.c
+++ b/drivers/clk/mvebu/armada-37xx-periph.c
@@ -638,10 +638,7 @@ static int armada_3700_add_composite_clk
 					mux_ops, rate_hw, rate_ops,
 					gate_hw, gate_ops, CLK_IGNORE_UNUSED);
 
-	if (IS_ERR(*hw))
-		return PTR_ERR(*hw);
-
-	return 0;
+	return PTR_ERR_OR_ZERO(*hw);
 }
 
 static int armada_3700_periph_clock_probe(struct platform_device *pdev)
